commands/auto.js

```diff
@@ -0,0 +1,44 @@
+const {article,proper} = require("../modules/lang");
+
+module.exports = {
+	help: cfg => "Set " + article(cfg) + " " + cfg.lang + " as your default " + cfg.lang + " for new messages.",
+	usage: cfg =>  ["auto <name> - Set the named " + cfg.lang + " as your auto " + cfg.lang],
+	permitted: () => true,
+	groupArgs: true,
+	execute: async (bot, msg, args, cfg) => {
+        console.log('config object');
+        console.log(cfg);
+        var client = null;
+        try{
+            client = await bot.db.connect();
+            //if no user given, clear the current auto user
+            if(!args[0]){
+                //delete the auto user and the given message
+                await bot.db.deleteAuto(msg.author.id, client);
+                let perms = msg.channel.permissionsOf(bot.user.id);
+                if (perms.has("manageMessages") && perms.has("readMessages"))
+                    process.send({ name: "queueDelete", channelID: msg.channel.id, messageID: msg.id }, null, { swallowErrors: false }, err => {
+                        if (err) console.log(err)
+                    });
+                return;
+            } 
+
+            //check arguments
+            let name = args.join(" ");
+            let member = await bot.db.getMember(msg.author.id,name);
+            if(!member) return "No valid " + cfg.lang + " found for the given name.";
+
+            //set the auto user and delete the given message
+            await bot.db.deleteAuto(msg.author.id,client);
+            await bot.db.addAuto(msg.author.id,member, client);
+            let perms = msg.channel.permissionsOf(bot.user.id);
+            if (perms.has("manageMessages") && perms.has("readMessages"))
+                process.send({ name: "queueDelete", channelID: msg.channel.id, messageID: msg.id }, null, { swallowErrors: false }, err => {
```

Comment:
Again, command messages shouldn't be deleted

Reply:
My main reasoning for this is that a person will want to quickly punch this in during a conversation, and removing them is to keep conversation from getting cluttered so users won't have to jump over to a dedicated bot channel just to change what they're doing.  It's a command, but it's going to be used in a context where most people will not want it around.

Reply:
To be fair, in a lot of servers they'd have to do that anyway due to command blacklisting.

As a personal note from someone who uses PK's autoproxy: having the command there really isn't a big deal. It doesn't interrupt conversation much and is easy to ignore. Not removing it also has the added benefit of working as rudimentary switch tracking since Tbox doesn't have that yet, as you can then search your own messages for where you used the command.

Reply:
I can set it either way if asked, but I do think it's best removed.